src/context-menu/context-menu.component.ts

```diff
@@ -0,0 +1,140 @@
+import {
+	Component,
+	ContentChildren,
+	ElementRef,
+	HostListener,
+	Input,
+	OnInit,
+	Optional,
+	SimpleChanges,
+	SkipSelf,
+	QueryList,
+	forwardRef,
+	OnChanges
+} from "@angular/core";
+import { ContextMenuItemComponent } from "./context-menu-item.component";
+
+@Component({
+	selector: "ibm-context-menu",
+	template: `
+		<ul
+			class="bx--context-menu"
+			[ngClass]="{
+				'bx--context-menu--root': root,
+				'bx--context-menu--open': open
+			}"
+			role="menu"
+			[attr.data-level]="level"
+			tabindex="-1"
+			[ngStyle]="{
+				'left.px': position.left,
+				'top.px': position.top
+			}">
+			<ng-content></ng-content>
+		</ul>
+	`
+})
+export class ContextMenuComponent implements OnInit, OnChanges {
+	@Input() root = true;
+	@Input() open = false;
+	@Input() position = {
+		left: 0,
+		top: 0
+	};
+
+	public level = 0;
+
+	// tslint:disable-next-line
+	@ContentChildren(forwardRef(() => ContextMenuItemComponent), { descendants: true }) menuItems: QueryList<ContextMenuItemComponent>;
+	// tslint:disable-next-line
+	@ContentChildren(forwardRef(() => ContextMenuComponent), { descendants: true }) subMenus: QueryList<ContextMenuComponent>;
+
+	constructor(
+		@SkipSelf() @Optional() protected parentContextMenu: ContextMenuComponent,
+		protected elementRef: ElementRef
+	) { }
+
+	ngOnChanges(changes: SimpleChanges) {
+		if (changes.open) {
+			if (changes.open.currentValue) {
+				this.focusMenu();
+			}
+		}
+	}
+
+	ngOnInit() {
+		if (this.parentContextMenu && this.parentContextMenu.root) {
+			this.root = false;
+			this.open = false;
+			this.level = this.parentContextMenu.level + 1;
+		}
+	}
+
+	focusMenu() {
+		// wait until the next tick to let the DOM settle before changing the focus
+		setTimeout(() => {
+			if (this.root) {
+				this.elementRef.nativeElement.querySelector("ul").focus();
+			} else {
+				this.menuItems.first.focusItem();
+			}
+		});
+	}
+
+	@HostListener("keydown", ["$event"])
+	handleNavigation(event: KeyboardEvent) {
+		const subMenus = this.subMenus.toArray().filter(subMenu => subMenu !== this);
+		const menuItems = this.menuItems.toArray().filter(menuItem => {
+			// if the menu item is contained within any submenu then remove it from the array
+			return !subMenus.some(subMenu => {
+				// check if the menu item is contained within a submenu
+				return subMenu.menuItems.some(subMenuItem => {
+					return menuItem === subMenuItem;
+				});
+			});
+		});
+		const currentIndex = menuItems.findIndex(menuItem => menuItem.tabindex === 0);
+		const list: HTMLElement = this.elementRef.nativeElement.querySelector("ul");
+
+		if (event.key === "ArrowDown") {
```

Comment:
Looks like you're trying to create a switch-case type flow with ifs on event.key. Maybe just use switch?